Tests/SwiftSyntaxTest/SyntaxTests.swift

```diff
@@ -168,4 +168,39 @@ class SyntaxTests: XCTestCase {
     XCTAssertEqual(funcKeywordInTree2?.as(TokenSyntax.self)?.tokenKind, .keyword(.func))
     XCTAssertNotEqual(funcKeywordInTree1.id, funcKeywordInTree2?.id)
   }
+
+  func testIntegerLiteralExprSyntax() {
+    let testCases: [UInt: (String, Int?)] = [
+      #line: ("2", 2),
+      #line: ("02", 2),
+      #line: ("020", 20),
+      #line: ("-02", -2),
+      #line: ("2_00_0000", 2000000),
+      #line: ("-2_00_0000", -2000000),
+      #line: ("foo", nil),
+      #line: ("999999999999999999999999999999999999999999999999999999999999999999999999999999", nil),
+      #line: ("0b1010101", 85),
+      #line: ("0xFF", 255),
+      #line: ("0o777", 511),
```

Comment:
What do you think about making the right-hand-side of this check be exactly the same as the left-hand side, just without the string literals? That way you can easily tell that `integerValue` and `floatingValue` getters parse the value in the same way as the parser during compilation time.

Reply:
Do you mean changing `#line: ("2_00_0000", 2_00_0000),` etc?

Reply:
Yes, exactly.